tests/unit/test_check_operators/test_containment_checks.py

```diff
@@ -511,3 +511,68 @@ def test_is_contained_by_with_none_in_comparison(
         {"target": "target", "comparator": comparator}
     )
     assert result.equals(df.convert_to_series(expected_result))
+
+
+@pytest.mark.parametrize(
+    "column_data,expected",
+    [
+        ([["A", "B"], ["C", "D"], ["E", "F"]], True),
+        ([{"A", "B"}, {"C", "D"}], True),
+        ([None, ["A", "B"], ["C", "D"]], True),
+        ([["A", "B"], None, ["C", "D"]], True),
+        ([["A", "B"], ["C", "D"], None], True),
+        ([None, None, ["A", "B"]], True),
+        ([None, {"A", "B"}, {"C", "D"}], True),
+        ([[]], True),
+        ([set()], True),
+        ([None, []], True),
+        ([None, set()], True),
+        ([["A"], []], True),
+        ([["A"], set()], True),
+        ([{"A"}, []], True),
+        ([["A"]], True),
+        ([{"A"}], True),
+        ([None, ["A"]], True),
+        ([["A"], {"B"}], True),
+        ([None, ["A"], {"B"}], True),
+        ([float("nan")], False),
+        ([None, float("nan")], False),
+        ([None, float("nan"), ["A"]], True),
+        ([float("nan"), ["A"]], True),
+        # Negative cases - not iterables
+        (["A", "B", "C"], False),
+        ([None, "A", "B"], False),
+        ([["A", "B"], "C", ["D", "E"]], False),
+        ([None, None, None], False),
+        ([], False),
+        ([("A", "B")], False),
+        ([None, ("A", "B")], False),
+        ([["A"], ("B", "C")], False),
+    ],
+)
+def test_is_column_of_iterables(column_data, expected):
+    df = PandasDataset.from_dict({"col": column_data})
+    dataframe_operator = DataframeType({"value": df})
+    result = dataframe_operator.is_column_of_iterables(df["col"])
+    assert result == expected
+
+
+def test_is_column_of_iterables_with_pd_na():
+    """Test pd.NA handling (pandas NA value)"""
+    import pandas as pd
+
+    test_cases = [
+        ([pd.NA], False),
+        ([None, pd.NA], False),
+        ([None, pd.NA, ["A"]], True),
+        ([pd.NA, ["A"]], True),
+        ([None, pd.NA, ["A"], ["B"]], True),
+    ]
```

Comment:
Sorry, one more thing. I think these test cases should be combined to a single function